### PR TITLE
Feature/mark orders as delivered

### PR DESCRIPTION
**Pull request by Gludek**

### CODE REVIEW

File: app/controllers/admins/orders_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      end
+    end
+
+    private
+
+    def order
+      @order ||= Order.find(params[:id])
+    end
+
+    def order_params
```
Comment (ciejj) [inline app/controllers/admins/orders_controller.rb:17]: is this method used?

File: app/models/order.rb [hunk 2]
```diff
@@ -1,4 +1,7 @@
 class Order < ApplicationRecord
   belongs_to :buyer, class_name: 'Employee'
   serialize :reward, JSON
+  enum status: { awaiting_delivery: 0, delivered: 1 }
+  scope :ordered_by_status, ->(ordering_method = 'asc') { order(status: ordering_method) }
+  scope :undelivered, -> { where(status: :awaiting_delivery) }
```
Comment (ciejj) [inline app/models/order.rb:6]: you already have that scope thanks to enum - https://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html

File: app/controllers/admins/orders_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+module Admins
+  class OrdersController < AdminsController
+    def update
```
Comment (ciejj) [inline app/controllers/admins/orders_controller.rb:3]: so update is not really updating, but setting status to delivered - I would pass some parameters here, or rename the action to deliver or something similar

File: app/views/admins/employees/index.html.erb [hunk 4]
```diff
@@ -14,7 +14,7 @@
         <td><%= employee.email %></td>
         <td><%= employee.number_of_available_kudos %></td>
         <td><%= employee.created_at.strftime("%d/%m/%Y") %></td>
-        <td><%= link_to "Profile", admins_employee_path(employee.id) %></td>
+        <td><%= link_to "Profile#{employee.orders.undelivered.size > 0 ? " (#{employee.orders.undelivered.size})" : ""}", admins_employee_path(employee.id) %></td>
```
Comment (ciejj) [inline app/views/admins/employees/index.html.erb:17]:
```suggestion
        <td><%= link_to "Profile#{employee.orders.undelivered.any? " (#{employee.orders.undelivered.size})" : ""}", admins_employee_path(employee.id) %></td>
```

File: spec/system/admins/order_handling_spec.rb [hunk 5]
```diff
@@ -18,4 +18,14 @@
     expect(page).to have_content reward.price.to_s
     expect(page).to have_content employee.email
   end
+
+  it 'Admin to deliver orders' do
```
Comment (ciejj) [inline spec/system/admins/order_handling_spec.rb:22]: It would be good to test admin trying deliver already delivered order

**Review by ciejj — APPROVED** [review]
👍 nice one!

**Comment by Gludek** [comment]
@ciejj could you please check if new commits solved suggested changes? I'm especially not sure about test that checks wheter admin can deliver already delivered order especially since it doesn't create link for delivered orders.

**Comment by ciejj** [comment]
Hey, @Gludek  changes look good. As regards additional checks - since this is the feature spec - I suggest checking that when the order is delivered the 'deliver' button is not available

**Pull request merged**